apis/middlewares.go

```diff
@@ -160,23 +160,25 @@ func LoadAuthContext(app core.App) echo.MiddlewareFunc {
 		return func(c echo.Context) error {
 			token := c.Request().Header.Get("Authorization")
 
-			if token != "" {
-				if strings.HasPrefix(token, "User ") {
-					user, err := app.Dao().FindUserByToken(
-						token[5:],
-						app.Settings().UserAuthToken.Secret,
-					)
-					if err == nil && user != nil {
-						c.Set(ContextUserKey, user)
-					}
-				} else if strings.HasPrefix(token, "Admin ") {
-					admin, err := app.Dao().FindAdminByToken(
-						token[6:],
-						app.Settings().AdminAuthToken.Secret,
-					)
-					if err == nil && admin != nil {
-						c.Set(ContextAdminKey, admin)
-					}
+			rolePrefix := "User "
+			if strings.HasPrefix(token, rolePrefix) {
+				user, err := app.Dao().FindUserByToken(
+					token[len(rolePrefix):],
+					app.Settings().UserAuthToken.Secret,
+				)
+				if err == nil && user != nil {
+					c.Set(ContextUserKey, user)
+				}
+			}
+
+			rolePrefix = "Admin "
+			if strings.HasPrefix(token, rolePrefix) {
+				admin, err := app.Dao().FindAdminByToken(
+					token[len(rolePrefix):],
+					app.Settings().AdminAuthToken.Secret,
+				)
+				if err == nil && admin != nil {
+					c.Set(ContextAdminKey, admin)
```

Comment:
This change may not be necessary, since I don't think there is a need to check the token prefix if is empty and 2 times in that. Let's keep the original (eventually we may want to return early to reduce the nesting, but the function is small and it doesn't hurt the readability too much)